lib/features/exercises/bloc/exercises_bloc.dart

```diff
@@ -24,4 +24,21 @@ class ExercisesBloc extends Cubit<ExercisesState> {
     final exercises = await _dataSource.loadExercises();
     emit(state.copyWith(exercises: exercises));
   }
+
+  void updateExercise(Exercise oldExercise, Exercise newExercise) {
+    final indexOldExercise = state.exercises.indexOf(oldExercise);
+    state.exercises.remove(oldExercise);
+    _dataSource.removeExercise(oldExercise);
+    state.copyWith(exercises: state.exercises);
+    state.exercises.insert(indexOldExercise, newExercise);
+    _dataSource.saveExercises(state.exercises);
+
+    emit(state.copyWith(exercises: state.exercises));
+  }
+
+  void deleteExercise(Exercise exercise) {
+    state.exercises.remove(exercise);
+    _dataSource.removeExercise(exercise);
```

Comment:
if will be exception in _dataSource? and you wont remove data from here but for state you already do it